### ISSUE TITLE
[BUG] Can't parse solution file

### ISSUE THREAD

**Issue by dmorozov-renuity** [microsoft/vscode-dotnettools#430]
### Describe the Issue

When opening any solution file it shows an error popup at the right/bottom which contains error log below.
The class resolution/code navigation stops working. Practically impossible to do development. I.e. you still can run both the application or tests.

```
===================
8/23/2023 4:46:24 PM
LimitedFunctionality
System.AggregateException: Project system data flow 'Workspace update handler 46609839' closed because of an exception: Invalid URI: The format of the URI could not be determined.. (Invalid URI: The format of the URI could not be determined.)
---> (Inner Exception #0) StreamJsonRpc.RemoteInvocationException: Invalid URI: The format of the URI could not be determined.
   at StreamJsonRpc.JsonRpc.InvokeCoreAsync[TResult](RequestId id, String targetName, IReadOnlyList`1 arguments, IReadOnlyList`1 positionalArgumentDeclaredTypes, IReadOnlyDictionary`2 namedArgumentDeclaredTypes, CancellationToken cancellationToken, Boolean isParameterObject)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass46_1`1.<<OnProjectChangedAsync>g__ApplyInBatchAsync|1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.<>c__DisplayClass3_0.<<ExecuteAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.ExecuteAsync(Func`1 operation, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.OnceInitializedOnceDisposedUnderLockAsync.ExecuteUnderLockCoreAsync(Func`2 action, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.OnBuildUpdateAsync(IProjectVersionedValue`1 update)
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass42_0.<<OnWorkspaceUpdateAsync>g__ApplyUpdateWithinLockAsync|1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass42_0.<<OnWorkspaceUpdateAsync>g__ApplyUpdateWithinLockAsync|1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.ProjectSystem.VSCode.LanguageServices.Workspace.<>c__DisplayClass42_0.<<OnWorkspaceUpdateAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.<>c__DisplayClass3_0.<<ExecuteAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.VisualStudio.Threading.ReentrantSemaphore.StackSemaphore.ExecuteAsync(Func`1 operation, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.OnceInitializedOnceDisposedUnderLockAsync.ExecuteUnderLockCoreAsync(Func`2 action, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.ProjectSystem.DataReceivingBlockSlim`1.ProcessInputQueueAsync(InputDataBuffer inputQueue)
<--- (Inner Exception #0) 
===================
```

### Steps To Reproduce

1. Open new window
2. In the Explorer panel left side click "Create .NET Project"
3. Choose ".NET Core Empty", select a folder, and provide the name of the new project.
4. After re-load it shows an error popup at the right / bottom.

I believe this is because it can't parse relative project folders referenced in the sln file.
As soon as I prefix the relative path with the "file:" prefix that issue is gone but of course the "dotnet restore" will not work anymore.
I.e.

from:
Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "WebApplication1", "WebApplication1\WebApplication1.csproj", "{FC1271D7-51BC-48D0-8631-B3E13F010A94}"
EndProject

to:
Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "WebApplication1", "file:\WebApplication1\WebApplication1.csproj", "{FC1271D7-51BC-48D0-8631-B3E13F010A94}"
EndProject

the error "Invalid URI" is gone.

The number of errors in the log file is equal to the number of sub-projects included in the solution. 

### Expected Behavior

No error when open the solution file.

### Environment Information

Linux (Ubuntu)

VSCode:
Version: 1.81.1
Commit: 6c3e3dba23e8fadc360aed75ce363ba185c49794
Date: 2023-08-09T22:18:39.991Z
Electron: 22.3.18
ElectronBuildId: 22689846
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Linux x64 6.2.0-26-generic snap

C# Dev Kit: 0.4.6

$ dotnet --info
.NET SDK:
 Version:   7.0.400
 Commit:    73bf45718d

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  22.04
 OS Platform: Linux
 RID:         ubuntu.22.04-x64
 Base Path:   /usr/share/dotnet/sdk/7.0.400/

Host:
  Version:      7.0.10
  Architecture: x64
  Commit:       a6dbb800a4

.NET SDKs installed:
  7.0.400 [/usr/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 7.0.10 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 7.0.10 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  Not set

global.json file:
  Not found

**Comment by dmorozov-renuity** [comment]
More details:
I tried to downgrade the C# Dev Kit extension first to the release version (3.x) and then to the older version 0.2.100
The same issue. Easy to reproduce with the simple steps with the empty project creation from the C# Dev Kit welcome walkthrough.

That looks like a global issue and not the C# Dev Kit extension.
I can only think of a recent update for the .NET SDK as the reason.
If you can guys give me any clue of how to fix it, I would appreciate. Kinda show stopper for me :(

**Comment by darkguy2008** [comment]
Not only a show stopper, it's a job stopper! I'm unable to do anything useful since this started happening.

Oh, how I miss the Omnisharp days, things used to work way better back then...

**Comment by dmorozov-renuity** [comment]
Yep, job stopper.

From the bright side, I think I can confirm that this issue is related to the .NET SDK version.
I.e. I ran the same project with the release version of "C# Dev Kit" v0.3.21 in the devcontainer where I can control the version of .NET SDK and it works.

I.e. no issues with .NET SDK 7.0.306   
The issue above related to the latest .NET SDK update (see version details in the environment information above).

**Comment by khrome666** [comment]
Not sure it is a .NET SDK problem...

I'm using .NET SDK version 7.0.110 (Linux Mint 21.2 Victoria). Yesterday I coded normally, no issues. Today I returned to the same solution, zero errors, updated the C# extension (not the C# Dev Kit, just the one extension), et voilá: error message, no syntax highlight on classes, no autocomplete.

I loved the entire VS Code environment, still do, but ever since MS took over, it's been a downhill trip...
